packages/vscode-extension/src/utilities/packageManager.ts

```diff
@@ -14,49 +18,74 @@ async function pathExists(p: string) {
   }
 }
 
-export async function resolvePackageManager(): Promise<PackageManagerName> {
-  const workspacePath = getAppRootFolder();
-  return await Promise.all([
-    pathExists(resolve(workspacePath, "yarn.lock")),
-    pathExists(resolve(workspacePath, "package-lock.json")),
-    pathExists(resolve(workspacePath, "pnpm-lock.yaml")),
-    pathExists(resolve(workspacePath, "bun.lockb")),
-  ]).then(([isYarn, isNpm, isPnpm, isBun]) => {
-    if (isYarn) {
-      return "yarn";
-    } else if (isPnpm) {
-      return "pnpm";
-    } else if (isBun) {
-      return "bun";
-    } else if (isNpm) {
-      return "npm";
+export async function resolvePackageManager(): Promise<PackageManagerInfo> {
+  function findWorkspace(appRoot: string) {
+    let currentDir = appRoot;
+    let parentDir = path.resolve(currentDir, "..");
+    while (parentDir !== currentDir) {
+      currentDir = parentDir;
+      parentDir = path.resolve(currentDir, "..");
+      const packageJsonPath = path.join(currentDir, "package.json");
+      if (!existsSync(packageJsonPath)) {
+        continue;
+      }
```

Comment:
I think we should just read the file and catch if doesn't exist, something like that:
```ts
function walkDirectoriesUntilRoot(
  start: string,
  shouldStop: (current: string) => boolean
): string | undefined {
  let currentDir = start;
  let parentDir = path.resolve(currentDir, "..");
  while (parentDir !== currentDir) {
    currentDir = parentDir;
    parentDir = path.resolve(currentDir, "..");

    if (shouldStop(currentDir)) {
      return currentDir;
    }
  }
  return undefined;
}

function findWorkspace(appRoot: string) {
  return walkDirectoriesUntilRoot(appRoot, (currentDir) => {
    try {
      const packageJsonPath = path.join(currentDir, "package.json");
      const hasWorkspaces = require(packageJsonPath).workspaces !== undefined;
      return hasWorkspaces;
    } catch (error) {
      // No package.json
      return false;
    }
  });
}
```

Reply:
i did that just reaarengedsome functions to reuse code